pkg/jobs/registry.go

```diff
@@ -895,6 +899,141 @@ func (r *Registry) Start(ctx context.Context, stopper *stop.Stopper) error {
 	})
 }
 
+func (r *Registry) createJobsStatsPollerIfNotExists(ctx context.Context) error {
+	if r.knobs.DisableJobsMetricsPolling {
```

Comment:
perhaps this knob can go away (if you change SHOW JOBS to not show this)

Reply:
I think we need this knob still. For example, `TestRegistryLifecycle` uses jobs testing knobs that block in `registry.stepThroughStateMachine`. It has to disable all background jobs. There's also `TestRegistryGC` which modifies `system.Jobs`, so it disables background jobs. There's also some tests which don't give their jobs user names (which causes crdb_internal.jobs to silently fail to decode the payload), so I added usernames to the job records.